File: composeApp/src/desktopMain/java/ies/java/model/CestaId.java
package ies.java.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import org.hibernate.Hibernate;

import java.io.Serializable;
import java.util.Objects;

@Embeddable
public class CestaId implements Serializable {
    private static final long serialVersionUID = -7826623052944399041L;
    @Column(name = "articulo", nullable = false, length = 7)
    private String articulo;

    @Column(name = "usuario", nullable = false, length = 50)
    private String usuario;

    public String getArticulo() {
        return articulo;
    }

    public void setArticulo(String articulo) {
        this.articulo = articulo;
    }

    public String getUsuario() {
        return usuario;
    }

    public void setUsuario(String usuario) {
        this.usuario = usuario;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        CestaId entity = (CestaId) o;
        return Objects.equals(this.usuario, entity.usuario) &&
                Objects.equals(this.articulo, entity.articulo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(usuario, articulo);
    }

}

File: composeApp/src/desktopMain/java/ies/java/model/CestaId.kt
package ies.java.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import org.hibernate.Hibernate
import java.io.Serializable
import java.util.*

@Embeddable
open class CestaId : Serializable {
    @Column(name = "articulo", nullable = false, length = 7)
    open var articulo: String? = null

    @Column(name = "usuario", nullable = false, length = 50)
    open var usuario: String? = null
    override fun hashCode(): Int = Objects.hash(articulo, usuario)
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false

        other as CestaId

        return articulo == other.articulo &&
                usuario == other.usuario
    }

    companion object {
        private const val serialVersionUID = -8120969969343397184L
    }
}